base/dates/parse.jl

```diff
@@ -50,8 +54,9 @@ function gen_exception(tokens, err_idx, pos)
     end
 end
 
-function reorder_args{Nv, Ni}(val::NTuple{Nv}, idx::NTuple{Ni}, default::NTuple{Ni}, valid_till)
-    ntuple(Val{Ni}) do i
+function reorder_args(val::Tuple, idx::Tuple, default::Tuple, valid_till::Integer)
+    # Note: idx and default have the same length
+    ntuple(length(idx)) do i
```

Comment:
@vtjnash any other way to facilitate the compiler figuring out the length of the tuple? I presume that uncertainty is why it needs to box this tuple?